specs/tooltip.md

```diff
@@ -0,0 +1,74 @@
+# Tooltip
+
+## Overview
+
+A tooltip is a popup that displays information related to an element when the element receives keyboard focus or the mouse hovers over it. It typically appears after a small delay and disappears when Escape is pressed or on mouse out.
+
+### Use Cases
+
+Tooltips are often used to supplement UI elements and provide additional information. Tooltips will often provide accessible names or add additional context to other UI components (such as icon buttons). While tooltips are natively provided to elements via the title attribute, they are notoriously difficult to style. Additionally, they are only invoked on hover and not on focus.
+
+### Non-goals
+Tooltip widgets do not receive focus. A hover that contains focusable elements can be made using a non-modal dialog (flyout).
+
+### Prior Art/Examples
+- [Lightning Design](https://www.lightningdesignsystem.com/components/tooltips/)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/tooltip)
+- [Evergreen](https://evergreen.segment.com/components/tooltip/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/tooltip/code/)
+- [Ant Design](https://ant.design/components/tooltip/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/tooltip)
+- [Bootstrap](https://getbootstrap.com/docs/4.3/components/tooltips/)
+
+---
+
+### API
+
+*Component name:*
+- `fast-tooltip`
+
+*Attributes:*
+- `anchor` - the html id of the HTMLElement which triggers the tooltip. This element references the tooltip element with aria-describedby.
+- `delay` - time in milliseconds to wait before showing and hiding the tooltip. Defaults to 300.
+- `visible` - the visiblity of the tooltip
+- `position` - enum; where the tooltip should appear relative to its target.
```

Comment:
Can this be based on the viewport positioner as well? It seems like a desired position would be good, but that we adjust if it's out of bounds.

Reply:
Per the PR description this work is dependent on Anchored Region :). We will not expose everything from Anchored Region though; this will start with very high-level and basic support to keep this positioned within the viewport.

Reply:
Should we have some migration plan to ensure that we can get to our richer api without needing to blow up the initial attributes?  Specifically - if we go to a tooltip that can be configured to always be in the same relative position to it's anchor or to choose where there is the most room what happens to position?

Reply:
I may be missing your question, but by not exposing those initially the API is additive and we won't blow up attrs or break folks.

Reply:
I guess my question is how do you plan to add to the API without needing to change the initial parts?  In this case how do we get to a place where there is a "position" attribute that is sometimes absolute (ie. always render above) and sometimes a preference (ie. render above, unless there isn't enough room)? For example in anchored region we use "default" position to communicate that the selection isn't always forced.

Reply:
I guess I'm not seeing an issue where adding more to a components API is problematic. In my mind, we start with scoped features. We know that a tooltip needs to be relative to its anchor or trigger so we offer options on where you'd like us to try and place it. If we think the positioning logic to keep it visible within the viewport is something that is valuable then that behavior starts as just baked. You can give us a position and we'll do our best to accommodate it so long as there is enough space. I don't think we need to bake every possible positioning feature in as part of an alpha release - we'll get a better idea of what people actually find helpful and useful when they start using it and filing issues they may come across.